src/routes/timelines.rs

```diff
@@ -0,0 +1,51 @@
+#![allow(dead_code)]
+
+use crate::{routes::errors::ErrorPage, template_utils::Ructe};
+use plume_models::{timeline::*, PlumeRocket};
+use rocket::response::Redirect;
+use routes::Page;
+use template_utils::IntoContext;
+
+#[get("/timeline/<id>?<page>")]
+pub fn details(id: i32, rockets: PlumeRocket, page: Option<Page>) -> Result<Ructe, ErrorPage> {
+    let page = page.unwrap_or_default();
+    let all_tl = Timeline::list_all_for_user(&rockets.conn, rockets.user.clone().map(|u| u.id))?;
+    let tl = Timeline::get(&rockets.conn, id)?;
+    let posts = tl.get_page(&rockets.conn, page.limits())?;
+    let total_posts = tl.count_posts(&rockets.conn)?;
+    Ok(render!(timelines::details(
+        &rockets.to_context(),
+        tl,
+        posts,
+        all_tl,
+        page.0,
+        Page::total(total_posts as i32)
+    )))
+}
+
+// TODO
```

Comment:
will it be done in this pr, or is this just basic support and actual customization will be added later?

Reply:
I will add it later I think, this PR is big enough.